Added documentation for online disk to brick resizing

ShyamsundarR opened the pull request:
Signed-off-by: ShyamsundarR <srangana@redhat.com>

JohnStrunk commented on docs/extend-EBS-size.md:
```diff
@@ -0,0 +1,62 @@
+# Extending Gluster volume size to accmodate EBS volume resize
+
+This documents details the operational procedure to expand a gluster volume on
```
s/documents/document/

JohnStrunk commented on docs/extend-EBS-size.md:
```diff
@@ -0,0 +1,62 @@
+relate to bricks and volumes setup using the ansible plays in this repository
+and do not form a general purpose instruction set for any Gluster instance.
+
+It is assumed that gluster brick deamon is up and running, while this exansion
```
expansion

JohnStrunk commented on docs/extend-EBS-size.md:
```diff
@@ -0,0 +1,62 @@
+    thin pool on a given VG in the setup
+    - Recheck LV size `lvs <vg-name>/pool` to ensure it has expanded
+
+5. Extend the thin LV backing the XFS instance
```
Let's break this into 2 different procedures:
- Enlarging the thin pool (1-4)
- Enlarging the active file system (5-8)

I suggest this because the overwrite rate will determine the appropriate relative size of active to total, and if that changes, we'll need to adjust them separately.

ShyamsundarR replied:
Ack

JohnStrunk reviewed (changes requested):
Please break this into 2 procedures, link it into the docs/README.md file, and squash.

ShyamsundarR commented:
Cross links between the docs work. @JohnStrunk  this is ready for review.

JohnStrunk commented on docs/expanding-lvm-thinpool.md:
```diff
@@ -0,0 +1,52 @@
+5. Repeat steps 1-4 on all nodes that form the Gluster volume
+
+## Example inputs:
+`<dev-path-to-disk>`: `/dev/xvdb`
```
This list needs some formatting help

ShyamsundarR replied:
Done

JohnStrunk commented on docs/resizing-gluster-brick.md:
```diff
@@ -0,0 +1,39 @@
+    - `df -kH <gluster-mountpoint>`
+
+## Example inputs:
+`<vg-name>`         : `supervol00` (can also be seen as volume name as that is
```
This list too.

ShyamsundarR replied:
done

JohnStrunk commented on docs/resizing-gluster-brick.md:
```diff
@@ -0,0 +1,39 @@
+2. Extend the thin LV backing the XFS instance
+    - Check and note the thin LV size `sudo lvs <vg-name>/<vg-name>`
+    - `sudo lvextend -L<size>  /dev/mapper/<vg-name>-<vg-name>`
+    **NOTE**: `<size>` is the total size that the LV needs to be thin
```
I find the phrasing here confusing. Perhaps sub-bullets under the lvextend:
- `<size>` is the **new size** for the LV, not the amount by which to extend the LV
- As of this writing, the LV for the active file system is typically sized at half of the thin pool size so that a 50% reserve is maintained for volume snapshots

ShyamsundarR replied:
done

JohnStrunk reviewed (changes requested):
- 2 minor formatting issues
- 1 phrasing
- Please add links in docs/README.md to these two new docs

ShyamsundarR commented on docs/expanding-lvm-thinpool.md:
```diff
@@ -0,0 +1,52 @@
+    - Check the current thin pool size `sudo lvs <vg-name>/pool`
+    - If LV size is smaller than the new VG size, proceed
+    - `sudo lvextend -l 100%VG /dev/mapper/<vg-name>-pool`
+    **NOTE**: We extend the thin pool to occupy entire VG as there is a single
```
NOTE here was not wrapping, corrected the same by removing the note and making it a bullet.

ShyamsundarR committed:
Added documentation for LVM thin pool and brick resizing

Added following operational procedures for,
- Resizing LVM thin pool
- Resizing Gluster brick

Both procedures are online resize procedures and tested with
Vagrant and AWS+EBS (gps) setups.

Signed-off-by: ShyamsundarR <srangana@redhat.com>
